pkg/alertmanager/merger/v1_alerts.go

```diff
@@ -0,0 +1,77 @@
+package merger
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+
+	v1 "github.com/prometheus/alertmanager/api/v1"
+)
+
+const (
+	statusSuccess = "success"
+)
+
+// V1Alerts implements the Merger interface for GET /v1/alerts. It returns the union of alerts over
+// all the responses. When the same alert exists in multiple responses, the alert instance in the
+// earliest response is returned in the final response. We cannot use the UpdatedAt timestamp as
+// for V2Alerts, because the v1 API does not provide it.
+type V1Alerts struct{}
+
+func (V1Alerts) MergeResponses(in [][]byte) ([]byte, error) {
+	type bodyType struct {
+		Status string      `json:"status"`
+		Data   []*v1.Alert `json:"data"`
+	}
+
+	alerts := make([]*v1.Alert, 0)
+	for _, body := range in {
+		parsed := bodyType{}
+		if err := json.Unmarshal(body, &parsed); err != nil {
+			return nil, err
+		}
+		if parsed.Status != statusSuccess {
```

Comment:
If all responses have status not-success, should we pick one and return it? It could be better than returning 5xx.

Reply:
My understanding of the distributor is that only successful responses get this far as they are checked earlier on. https://github.com/cortexproject/cortex/blob/master/pkg/alertmanager/distributor.go#L171